### ISSUE TITLE
redraw row data after add/edit

### ISSUE THREAD

**Issue by anasred** [fooplugins/FooTable#614]
I asked you before on how to remove row data from footable successful ajax delete call and you asked me to do this:
```javascript
if (row instanceof FooTable.Row){
	row.delete(true);
}
```

now i'm wondering what should i have to call after successful add or update ajax call?

I tried:
row.edit(true);
row.update(true);

but I got this reply on console:
row.edit is not a function / row.update is not a function

**Comment by steveush** [comment]
Hi,

I suggest taking a look at the built in [editing components example](http://fooplugins.github.io/FooTable/docs/examples/component/editing.html), all the basic operations are covered there. The built in editing component only creates the column and the buttons to initiate an edit operation and exposes callbacks to implement the logic to do so, the logic is all written by you, the developer, and so the example shows how to add/edit/delete as per your original questions.

At the moment it seems like you are duplicating having to create the UI buttons which is what the built in component does. As you can see in the example you provide the Bootstrap modal to display as you are doing now, the logic simply moves from the previously (#612) supplied delegated `click` event to the `deleteRow( row )` callback with the row object is supplied as the first parameter for you. The callbacks supplied by the editing component can basically be thought of as `click` events for those various actions. So the implementation logic is under your control, you are just simply provided the `row` object as the first parameter when required.

To summarize what is in the example though, given a simple table with 3 columns; id, name and jobTitle:

Add Row:

```javascript
var ft = /* Instance of FooTable.Table */;
ft.rows.add({
	id: 123,
	name: "Steve",
	jobTitle: "Developer"
});
```

Delete Row:

```javascript
var row = /* Instance of FooTable.Row */;
row.delete();
```

Update Row (Not all column values are required each time, the `row.val()` method handles partial updates):

```javascript
var row = /* Instance of FooTable.Row */;
row.val({name: "Steve"});
```

If you are still unclear let me know.

Thanks

**Comment by anasred** [comment]
I tried a lot, but i couldn't force the table row to refresh after calling row.value() and adds the new passed data.

here is my full code

```javascript
$("#contacts").on("click", ".edit-btn", function(event){
    var row = FooTable.getRow(this)

    $('#edit').on('show.bs.modal', function(event) {
        var button = $(event.relatedTarget)
	var id = button.data('id')
	var modal = $(this)

        //I have contacts variable which passed from my controller to the view
	var name = modal.find('.modal-body input#name').val(contacts[id].name)
	var positions = modal.find('.modal-body input#position').val(contacts[id].position)
	var email = modal.find('.modal-body input#email').val(contacts[id].email)
	var mobile = modal.find('.modal-body input#mobile').val(contacts[id].mobile)
	var telephone = modal.find('.modal-body input#telephone').val(contacts[id].telephone)
	var fax = modal.find('.modal-body input#fax').val(contacts[id].fax)

	    $("button#edit").click(function(){
                $.ajax({
			type: "patch",
			url: "/contact/" + id,
			data: {
				'id': id,
				'name': modal.find('.modal-body input#name').val(),
				'position': modal.find('.modal-body input#position').val(),
				'email': modal.find('.modal-body input#email').val(),
				'mobile': modal.find('.modal-body input#mobile').val(),
				'telephone': modal.find('.modal-body input#telephone').val(),
				'fax': modal.find('.modal-body input#fax').val(),
			},
			success: function(data){
				if (row instanceof FooTable.Row){
					row.val(data[0])
				}
				$('#edit').modal('hide')
			},
			error: function(){
				alert("Error")
			}
		});
        });
    });
});
```

please help out with this.

**Comment by steveush** [comment]
Hi @anasred,

Everything looks correct in your code, the only thing I can think of is the supplied parameter to the success callback is not what is expected? Is the `data[0]` value an object whose properties match your column names?

For example with the following columns:

```javascript
columns: [
	{name: "id", type: "number"},
	{name: "name"},
	{name: "jobTitle"}
]
```

You `data[0]` object should look like:

```javascript
{
	id: 123,
	name: "Steve",
	jobTitle: "Developer"
}
```

Thanks
